### ISSUE TITLE
Emu crash: panic: counter can't be zero

### ISSUE THREAD

**Issue by syaakov** [cisco-system-traffic-generator/trex-emu#22]
trex-emu crashes when we run it with kernel-mode, with the error message: `panic: counter can't be zero`

output:
```
./trex-emu -k
Run ZMQ server on [RPC:4510, RX: IPC, TX:IPC]
panic: counter can't be zero

goroutine 1 [running]:
emu/core.(*cHTimerBucket).decCount(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/timerw.go:131
emu/core.(*CHTimerObj).detach(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/timerw.go:176
emu/core.(*cHTimerOneWheel).stop(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/timerw.go:294
emu/core.(*CNATimerWheel).Stop(0x895946?, 0xc0001c01d8?)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/timerw.go:542 +0xfb
emu/core.(*TimerCtx).Stop(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/timer_ctx.go:81
emu/plugins/arp.(*ArpFlowTable).MoveToLearn(0xc0000b41a0, 0xc00034a480)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/plugins/arp/arp.go:411 +0x45
emu/plugins/arp.(*PluginArpNs).DisassociateClient(0xc0000b4160, 0xc0000e61e0, {0x1, 0x1, 0x1, 0x1})
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/plugins/arp/arp.go:843 +0xd8
emu/plugins/arp.(*PluginArpClient).OnChangeDGSrcIPv4(0xc0000e61e0, {0x1, 0x1, 0x1, 0x1}, {0x1, 0x1, 0x1, 0x1}, 0xbd?, ...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/plugins/arp/arp.go:714 +0x59
emu/plugins/arp.(*PluginArpClient).OnRemove(0xc0000e61e0, 0xc0002fcb70)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/plugins/arp/arp.go:728 +0x93
emu/core.(*PluginCtx).RemovePlugins(0xc0002fcb70, {0xc00024cfe8, 0x3})
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/plugin_ctx.go:224 +0x86
emu/core.(*PluginCtx).OnRemove(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/plugin_ctx.go:171
emu/core.(*CClient).OnRemove(0x953180)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/client_ctx.go:200 +0xb8
emu/core.(*CNSCtx).RemoveClient(0xc0002086c0, 0xc000192200)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/ns_ctx.go:393 +0x139
emu/core.ApiClientRemoveHandler.ServeJSONRPC({}, {0x9e3800?, 0xc00022a000?}, 0xc0002ac650?)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/rpc_base_cmds.go:397 +0x173
external/osamingo/jsonrpc.(*MethodRepository).InvokeMethod(0xc0002aa280, {0xba?, 0x200?}, 0xc000842060)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/external/osamingo/jsonrpc/handler.go:135 +0x202
external/osamingo/jsonrpc.(*MethodRepository).ServeBytes(0xc0002aa280, {0xc000340000, 0xba, 0x200})
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/external/osamingo/jsonrpc/handler.go:53 +0x3e8
external/osamingo/jsonrpc.(*MethodRepository).ServeBytesCompress(0xc00006be10?, {0xc00033e000?, 0x410d25?, 0x0?})
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/external/osamingo/jsonrpc/handler.go:77 +0x65
emu/core.(*CZmqJsonRPC2).HandleReqToChan(...)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/rpc.go:116
emu/core.(*CThreadCtx).MainLoop(0xc00022a000)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/emu/core/thread_ctx.go:389 +0x147
main.RunCoreZmq(0xc00007e480)
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/cmd/trex-emu.go:183 +0x48c
main.main()
/tmp/build_838_TRex_Emu_1d2de09906f0266741b2c59c2f9e2c6fa287b809/src/cmd/trex-emu.go:191 +0x1e
```

**Comment by hhaim** [comment]
@obarash 
@syaakov 
in kernel mode there should not be a ARP plugin

**Comment by syaakov** [comment]
solved in [8a77593](https://github.com/cisco-system-traffic-generator/trex-emu/commit/8a77593e7ccc408812b0caa94c0d9341adf6c392)
